azuread/table_azuread_service_principal.go

```diff
@@ -19,159 +23,272 @@ func tableAzureAdServicePrincipal() *plugin.Table {
-			{Name: "sign_in_audience", Type: proto.ColumnType_STRING, Description: "Specifies the Microsoft accounts that are supported for the current application. Supported values are: AzureADMyOrg, AzureADMultipleOrgs, AzureADandPersonalMicrosoftAccount, PersonalMicrosoftAccount."},
-
-			// // Json fields
-			{Name: "add_ins", Type: proto.ColumnType_JSON, Description: "Defines custom behavior that a consuming service can use to call an app in specific contexts."},
-			{Name: "alternative_names", Type: proto.ColumnType_JSON, Description: "Used to retrieve service principals by subscription, identify resource group and full resource ids for managed identities."},
-			{Name: "app_roles", Type: proto.ColumnType_JSON, Description: "The roles exposed by the application which this service principal represents."},
-			{Name: "info", Type: proto.ColumnType_JSON, Description: "Basic profile information of the acquired application such as app's marketing, support, terms of service and privacy statement URLs."},
-			{Name: "keyCredentials", Type: proto.ColumnType_JSON, Description: "The collection of key credentials associated with the service principal."},
-			{Name: "notification_email_addresses", Type: proto.ColumnType_JSON, Description: "Specifies the list of email addresses where Azure AD sends a notification when the active certificate is near the expiration date. This is only for the certificates used to sign the SAML token issued for Azure AD Gallery applications."},
-			{Name: "owner_ids", Type: proto.ColumnType_JSON, Hydrate: getAdServicePrincipalOwners, Transform: transform.FromValue(), Description: "Id of the owners of the application. The owners are a set of non-admin users who are allowed to modify this object."},
-			{Name: "password_credentials", Type: proto.ColumnType_JSON, Description: "Represents a password credential associated with a service principal."},
-			{Name: "published_permission_scopes", Type: proto.ColumnType_JSON, Description: "The published permission scopes."},
-			{Name: "reply_urls", Type: proto.ColumnType_JSON, Description: "The URLs that user tokens are sent to for sign in with the associated application, or the redirect URIs that OAuth 2.0 authorization codes and access tokens are sent to for the associated application."},
-			{Name: "service_principal_names", Type: proto.ColumnType_JSON, Description: "Contains the list of identifiersUris, copied over from the associated application. Additional values can be added to hybrid applications. These values can be used to identify the permissions exposed by this app within Azure AD."},
-			{Name: "tags_src", Type: proto.ColumnType_JSON, Description: "Custom strings that can be used to categorize and identify the service principal.", Transform: transform.FromField("Tags")},
-			{Name: "verified_publisher", Type: proto.ColumnType_JSON, Description: "Specifies the verified publisher of the application which this service principal represents."},
```

Comment:
Did the old SDK return data for this column, if so do you have an example of that data? Also, does the base Graph API still return this data?

Reply:
Old SDK returns empty object `{}` for this column. I have a discussion with the MSGraph SDK team https://github.com/microsoftgraph/msgraph-sdk-go/issues/228#issuecomment-1206680466, and this property is not supposed to work for ServicePrincipals.